spinoso-string/src/lib.rs

```diff
@@ -1415,7 +1250,8 @@ impl String {
     /// [`String#b`]: https://ruby-doc.org/core-2.6.3/String.html#method-i-b
     #[inline]
     pub fn make_binary(&mut self) {
-        self.encoding = Encoding::Binary;
+        let old = self.inner.as_slice().to_vec();
+        self.inner = EncodedString::new(old, Encoding::Binary);
```

Comment:
I believe you mentioned that this implementation is incorrect given the docs for `String#b`? Do you remember the details enough to file a ticket?

Reply:
Yes, I'm hoping this is the issue I was thinking of:

```irb
irb(main):001:0> a = "a"
=> "a"
irb(main):002:0> b = a.b
=> "a"
irb(main):003:0> a.object_id
=> 18440
irb(main):004:0> b.object_id
=> 25120
```

Specifically: https://ruby-doc.org/core-3.1.1/String.html#method-i-b <-- Says it'll return a copied string, where as this Rust implementation is just changing the encoding still. If that still feels like the implementation isn't "correct", then I'll raise the issue!

Reply:
Thank you! See:

- https://github.com/artichoke/artichoke/issues/1720